### ISSUE TITLE
4.12.0: Test `tests/flambda/afl_lazy.ml` fails with msvc

### ISSUE THREAD

**Issue by bschommer** [ocaml/ocaml#10343]
The tests fails when OCaml is build with msvc with the following output:

```console
Running tests from 'tests/flambda' ...
 ... testing 'afl_lazy.ml' with 1 (flambda) => passed
 ... testing 'afl_lazy.ml' with 1.1 (native) => failed (Compiling program C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\afl_lazy.opt.exe from modules  afl_lazy.ml: command
C:\ocaml\src\runtime\ocamlrun.exe C:\ocaml\src\ocamlopt.exe  -I C:\ocaml\src\runtime  -nostdlib -I C:\ocaml\src\stdlib     -O3 -afl-instrument  -o C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\afl_lazy.opt.exe   afl_lazy.ml 
failed with exit code 2)
> Specified modules: afl_lazy.ml
> Source modules: afl_lazy.ml
> Running test flambda with 1 actions
> 
> Running action 1/1 (flambda)
> Action 1/1 (flambda) passed (support for flambda enabled)
> Running test native with 8 actions
> 
> Running action 1/8 (setup-ocamlopt.byte-build-env)
> Action 1/8 (setup-ocamlopt.byte-build-env) passed
> 
> Running action 2/8 (ocamlopt.byte)
> Compiling program C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\afl_lazy.opt.exe from modules  afl_lazy.ml
> Commandline: C:\ocaml\src\runtime\ocamlrun.exe C:\ocaml\src\ocamlopt.exe -I C:\ocaml\src\runtime -nostdlib -I C:\ocaml\src\stdlib -O3 -afl-instrument -o C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\afl_lazy.opt.exe afl_lazy.ml
>   Redirecting stdout to C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\ocamlopt.byte.output 
>   Redirecting stderr to C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\ocamlopt.byte.output 
> ### begin stdout ###
> ** Cannot resolve symbols for afl_lazy.obj:
>  caml_afl_area_ptr
>  caml_afl_prev_loc
> File "caml_startup", line 1:
> Error: Error during linking (exit code 2)
> ### end stdout ###
> Action 2/8 (ocamlopt.byte) failed (Compiling program C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\afl_lazy.opt.exe from modules  afl_lazy.ml: command
> C:\ocaml\src\runtime\ocamlrun.exe C:\ocaml\src\ocamlopt.exe  -I C:\ocaml\src\runtime  -nostdlib -I C:\ocaml\src\stdlib     -O3 -afl-instrument  -o C:\ocaml\src\testsuite\_ocamltest\tests/flambda\afl_lazy\ocamlopt.byte\afl_lazy.opt.exe   afl_lazy.ml 
```

**Comment by dra27** [comment]
This is #10107, which was intended for 4.12 but got missed. I've just cherry-picked it, so it will be fixed in 4.12.1 (the actual bug has been presented since AFL was added)

**Comment by bschommer** [comment]
Thanks for the fast reply.
